xarray/core/dataset.py

```diff
@@ -2750,6 +2751,8 @@ def reduce(self, func, dim=None, keep_attrs=False, numeric_only=False,
             Dataset with this object's DataArrays replaced with new DataArrays
             of summarized data and the indicated dimension(s) removed.
         """
+        if dim == ALL_DIMS:
```

Comment:
I usually prefer using `is` for comparison to sentinel objects -- it looks a little more idiomatic, and is more similar to what we do for comparison to `None`.